### ISSUE TITLE
Support Device Authentication flow for VS Code editor

### ISSUE THREAD

**Issue by RomanNikitenko** [eclipse/che#22139]
### Is your enhancement related to a problem? Please describe

It looks like some VS Code extensions requires authentication based on the Device Authentication flow.
Currently it's not supported on the Che-Code side.
So, the extension fails with the error: "User not authorized"
 

### Describe the solution you'd like

Add authentication based on the Device Authentication flow

### Additional context

[Device Authentication flow](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps#device-flow)

**Comment by RomanNikitenko** [comment]
User gets a new token when uses `Device Authentication flow`. I'm looking for a way how to store the token to a secret.
 `@kubernetes/client-node` is used to read/write a secret.
 I noticed that I have `HTTP request failed` error when I use `this.k8sConfig.loadFromCluster()` to get `git-credentials` secret. I don't have such error when I use `this.k8sConfig.loadFromDefault()`.
 
 The cause of the problem is: SA permissions.
 I've tested with Angel [this draft](https://github.com/devfile/devworkspace-operator/pull/1165) to check it ^. 
 
 So, should we extend workspace SA permissions to all secrets/configmaps in namespace? 
 or 
 does it bring potentially a problem?
 
 @l0rd could you share your opinion?

**Comment by l0rd** [comment]
> So, should we extend workspace SA permissions to all secrets/configmaps in namespace

That's fine.

**Comment by RomanNikitenko** [comment]
> > So, should we extend workspace SA permissions to all secrets/configmaps in namespace
> 
> That's fine.

thank you, Mario!
if so, I propose to open for review https://github.com/devfile/devworkspace-operator/pull/1165#issuecomment-1723367727

Within testing my pull request for the Device Authentication flow I've faced another problem: the workspace is restarted and loses websocket connection when `git-credentials` secret is created. I'm able to reproduce the problem without my changes, so I've filed a separate issue: https://github.com/eclipse/che/issues/22522.
